StoreBroker/Helpers.ps1

```
@@ -838,3 +838,48 @@ function Resolve-UnverifiedPath
         return $resolvedPath.ProviderPath
     }
 }
+
+function Ensure-Directory
+{
+<#
+    .SYNOPSIS
+        A utility function for ensuring a given directory exists.
+
+    .DESCRIPTION
+        A utility function for ensuring a given directory exists.
+
+        If the directory does not already exist, it will be created.
+
+    .PARAMETER Path
+        A full or relative path to the directory that should exist when the function exits.
+
+    .NOTES
+        Uses the Resolve-UnverifiedPath function to resolve relative paths.
+#>
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory)]
+        [string] $Path
+    )
+
+    try
+    {
+        $resolvedPath = Resolve-UnverifiedPath -Path $Path
+        if ($null -ne $resolvedPath)

```

What's the scenario when this will be `$null`? (besides of course the input value being `$null`)

---

This came about from testing with -ErrorAction Ignore, as explained above.  I can remove the check.

---

We really shouldn't be in a situation where `Resolve-UnverifiedPath` is returning `$null`.  If it did, I'm not sure that just silently falling back to using the original `$Path` is the best idea....it just means that the caller is not guaranteed that they'll be getting their file written to the expected location...especially since the .zip file is going to run into trouble since it also depends on a valid value coming back from `Resolve-UnverifiedPath`